File: push/src/main/java/com/xiaomi/slim/Ping.java
package com.xiaomi.slim;

import java.nio.ByteBuffer;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/slim/Ping.class */
public final class Ping extends Blob {
    public Ping() {
        setCmd(Blob.CMD_PING, null);
        setPacketID(Blob.CLIENT_PING_ID);
        setChannelId(0);
    }

    @Override // com.xiaomi.slim.Blob
    public int getSerializedSize() {
        if (getPayload().length == 0) {
            return 0;
        }
        return super.getSerializedSize();
    }

    @Override // com.xiaomi.slim.Blob
    ByteBuffer toByteArray(ByteBuffer byteBuffer) {
        return getPayload().length == 0 ? byteBuffer : super.toByteArray(byteBuffer);
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/slim/Ping.kt
package com.xiaomi.slim

import java.nio.ByteBuffer

/*
 * Stock reference: com.xiaomi.xmsf 7.4.67-C (versionCode 70004067),
 * split-XiaomiServiceFrameworkCN-master.apk sha256 444e9f128591e04e38672bfe44a246ab3fa97ae68e95882839d8a7afe766df2b,
 * JADX path: com.xiaomi.xmsf/stock/split-XiaomiServiceFrameworkCN-master/sources/pa/h.java
 * Stock class name is obfuscated as pa.h; this file keeps the deobfuscated com.xiaomi.slim.Ping API.
 */
class Ping : Blob() {
    init {
        setCmd(Blob.CMD_PING, null)
        packetID = Blob.CLIENT_PING_ID
        setChannelId(0)
    }

    override val serializedSize: Int
        get() = if (payload.isEmpty()) 0 else super.serializedSize

    override fun toByteArray(byteBuffer: ByteBuffer?): ByteBuffer {
        return if (payload.isEmpty()) {
            byteBuffer ?: ByteBuffer.allocate(0)
        } else {
            super.toByteArray(byteBuffer)
        }
    }
}
